### DISCUSSION TITLE
Conflict between Element Connection and SEOmatic sitemap settings

### DISCUSSION THREAD

**Discussion by strandofgenius** [solspace/craft-freeform#546] [Questions & Answers]
Queue jobs are running to generate a sitemap when an entry type has public URLs but the sitemap setting is set to disabled in the SEOmatic content SEO settings.

This only happens when someone fills out the form from the front end of the site and the Element Connection is made between Freeform and creating an entry.

The sitemap does not get added to sitemap.xml but it is going through a process to generate it. This is unnecessary and it seems to get stuck from time to time and bogs things down. This especially becomes problematic since the client has thousands of leads in the system and each time a lead fills out the form it creates a new job to generate the sitemap again.

Here is the initial discussion about this issue: https://github.com/nystudio107/craft-seomatic/issues/1212

**Steps to reproduce**

1. Create an entry type with a URI format and template
2. Go to SEOmatic > Content SEO > Entry Type > Sitemap settings
3. Disable all toggles related to the sitemap
4. Fill out the form
5. Entry is created via Freeform Element Connection Event
6. Queue appears to create a job to generate a sitemap for the entry type

**Expected behavior**
It is expected that because the entry type sitemap setting is disabled it would not create a queue job to generate a sitemap.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.53.1
- Freeform Version: 3.13.18
- Freeform Edition: Pro
- SEOmatic Version: 3.4.37
- Fresh Install or Upgrade: N/A

**Comment by kjmartens** [comment]
Sorry for the delay @strandofgenius...

I will see if there's anything we can do to prevent Freeform from triggering this queue job. 🙂

**Reply by strandofgenius** [reply]
That would be great! Keep me posted. Thanks.

**Reply by kjmartens** [reply]
We've replied to the issue in SEOmatic as we believe it might be an issue on their end. 🙂

**Comment by strandofgenius** [comment]
It looks as though a reply came back and it might be something on your end again. Any update on this?

**Reply by kjmartens** [reply]
Sorry for the delay @strandofgenius. This slipped through the cracks, but I've now created an internal issue item to have it revisited shortly. 🙂
